app/src/main/java/org/opencord/aaa/impl/StateMachine.java

```diff
@@ -558,14 +622,57 @@ public void logoff() {
         private final Logger log = getLogger(getClass());
         private String name = "UNAUTHORIZED_STATE";
 
+        @Override
         public void start() {
             log.info("Moving from UNAUTHORIZED state to STARTED state.");
         }
 
+        @Override
         public void logoff() {
             log.info("Moving from UNAUTHORIZED state to IDLE state.");
         }
     }
 
 
+    /**
+     * Class for cleaning the StateMachine for those session for which no response is coming--implementing timeout.
+     */
+    class CleanupTimerTask implements Runnable {
+        private final Logger log = getLogger(getClass());
+        private String sessionId;
+        private AaaManager aaaManager;
+        CleanupTimerTask(String sessionId, AaaManager aaaManager) {
+            this.sessionId = sessionId;
+            this.aaaManager = aaaManager;
+        }
+
+        @Override
+        public void run() {
+            StateMachine stateMachine = StateMachine.lookupStateMachineBySessionId(sessionId);
+            if (null != stateMachine) {
+                boolean noTrafficWithinThreshold = System.currentTimeMillis() -
+                        stateMachine.getlastEapolRadiusPacketReceivedTime() >
+                (cleanupTimerTimeOutInMins * 60 * 60) / 2;
```

Comment:
Why multiplying twice by 60? Also, put braces to clearly indicate the expression being evaluated

Reply:
Resolved